plugin/backend.go

```diff
@@ -107,16 +113,16 @@ func (b *GcpAuthBackend) IAMClient(s logical.Storage) (*iam.Service, error) {
 // https://pkg.go.dev/google.golang.org/api@v0.45.0/iamcredentials/v1#pkg-overview
 //
 // The client is cached.
-func (b *GcpAuthBackend) IAMCredentialsClient(s logical.Storage) (*iamcredentials.Service, error) {
-	httpClient, err := b.httpClient(s)
+func (b *GcpAuthBackend) IAMCredentialsClient(ctx context.Context, s logical.Storage) (*iamcredentials.Service, error) {
+	opts, err := b.clientOptions(ctx, s, "")
```

Comment:
This is only used in test code to sign a JWT. That's why there is no custom endpoint being passed here. I'm thinking it'd be best to move this over into `path_login_test.go` where it's used.

Reply:
I moved this to `path_login_test.go` in 845d864. I also moved some other code there that was only being used in tests.